### PR TITLE
VariantSet metadata not correctly filled

### PR DESCRIPTION
**Pull request by pcingola**
Solves issue #677

### CODE REVIEW

File: ga4gh/datamodel/variants.py [hunk 1]
```diff
@@ -509,14 +517,17 @@ def buildMetadata(
             metadata = protocol.VariantSetMetadata()
             metadata.key = key
             metadata.value = value
-            metadata.id = id
             metadata.type = type
             metadata.number = number
             metadata.description = description
+            if id == '':
```
Comment (dcolligan) [inline ga4gh/datamodel/variants.py]: try and avoid build-in names as variable names... consider using `id_` instead

File: tests/unit/test_compound_ids.py [hunk 2]
```diff
@@ -311,3 +311,12 @@ def testExperimentParse(self):
         self.assertEqual(cid.readGroup, "c")
         self.assertEqual(cid.experiment, "d")
         self.verifyParseFailure(idStr, datamodel.ExperimentCompoundId)
+
+    def testVariantSetMetadataCompoundIdParse(self):
```
Comment (dcolligan) [inline tests/unit/test_compound_ids.py]: there should be two tests functions for each new compound id... see the analogues in the file

File: ga4gh/datamodel/variants.py [hunk 3]
```diff
@@ -509,14 +517,17 @@ def buildMetadata(
...
+                id = self.getMetadataId(metadata)
+            metadata.id = id
             return metadata
 
         ret = []
         header = varFile.header
+        # TODO: Call VariantSetMetadataCompoundId
```
Comment (dcolligan) [inline ga4gh/datamodel/variants.py]: aren't you doing this in `getMetadataId`?

File: ga4gh/datamodel/variants.py [hunk 4]
```diff
@@ -509,14 +517,17 @@ def buildMetadata(
             metadata = protocol.VariantSetMetadata()
             metadata.key = key
             metadata.value = value
-            metadata.id = id
             metadata.type = type
```
Comment (dcolligan) [inline ga4gh/datamodel/variants.py]: I know you didn't add this line, but change `type` to `type_` while you're at it

File: ga4gh/datamodel/variants.py [hunk 5]
```diff
@@ -493,6 +493,14 @@ def getVariants(self, referenceName, startPosition, endPosition,
     def getMetadata(self):
         return self._metadata
 
+    def getMetadataId(self, metadata):
+        """
+        Returns the id of a metadata
+        """
+        id = str(datamodel.VariantSetMetadataCompoundId(
```
Comment (dcolligan) [inline ga4gh/datamodel/variants.py]: change `id` to `id_`

**Comment by dcolligan** [comment]
Good start but there are a few things that need fixing

File: ga4gh/datamodel/variants.py (hunk 1 above)
Comment (pcingola) [inline ga4gh/datamodel/variants.py]: This name is form the original code, I'll fix it.

File: ga4gh/datamodel/variants.py (hunk 3 above)
Comment (pcingola) [inline ga4gh/datamodel/variants.py]: Forgot to remove the comment. Thank you.

**Comment by pcingola** [comment]
Thank you for the review. I think I addressed your comments. 
If it looks OK to you, I'll squash commits.

File: ga4gh/datamodel/variants.py (hunk 3 above)
Comment (dcolligan) [inline ga4gh/datamodel/variants.py]: this comment still needs to be removed

Comment (pcingola) [inline ga4gh/datamodel/variants.py]: I have no idea how it got back in....

**Comment by dcolligan** [comment]
yep, looks good to me

**Comment by dcolligan** [comment]
+1

**Comment by jeromekelleher** [comment]
+1; merging. Thanks @pcingola!

**Pull request merged**